Cosminis/DataAccess/FriendsRepo.cs

```diff
@@ -0,0 +1,457 @@
+using DataAccess.Entities;
+using CustomExceptions;
+using Models;
+using System.Data.SqlClient;
+using DataAccess;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+
+namespace DataAccess;
+ 
+public class FriendsRepo : IFriendsDAO
+{
+    private readonly wearelosingsteamContext _context;
+    private readonly IUserDAO _userRepo;
+
+    public FriendsRepo(wearelosingsteamContext context, IUserDAO userRepo)
+    {
+        _context = context;
+        _userRepo = userRepo;
+    }
+
+    public List<Friends> GetAllRelationships()
+    {
+        return _context.Friends.ToList();
+    }
+
+    public List<Friends> ViewAllFriends(int userIdToLookup)
+    {
+        User userToLookup = _context.Users.Find(userIdToLookup);
+
+        IEnumerable<Friends> friendsQuery =
+            (from Friends in _context.Friends
+            where (Friends.UserIdTo == userToLookup.UserId) || (Friends.UserIdFrom == userToLookup.UserId)
+            select Friends);
+
+        if(friendsQuery == null)
```

Comment:
What is this checking? I thought friendsQuery simply stores the query parameters and does not execute until you do .ToList() later on

Reply:
Idk if you mean line 33 or 36... 

For 33, any user could be either the UserIdTo or UserIdFrom, so I am getting every possible relationship this user is a part of. 36 returns ResourceNotFound if they aren't in any relationships.